src/httpserver/routing/router.clj

```diff
@@ -0,0 +1,15 @@
+(ns httpserver.routing.router)
+
+(defn check-route [method path route]
+  (and (= (:method route) method) (= (:path route) path)))
+
+(defn find-route [method path routes]
+  (some #(when (check-route method path %) %) routes))
+
+(defn get-handler [request routes]
+  (:handler (find-route (:method request) (:path request) routes))
+  )
+
+(defn add-routes [server list-of-routes main-handler]
+  (doseq [route list-of-routes]
```

Comment:
I think you could use `run!` here ([documentation](https://clojuredocs.org/clojure.core/run!))